Update the serverless installation guide for Go (#11414)
DataDog/documentation: content/en/serverless/installation/go.md

If you previously set up Datadog Serverless using the Datadog Forwarder, see the [installation instructions][2].

> If your Go Lambda functions are still using runtime [go1.x], consider either [migrating][3] to `provided.al2` or using the [Datadog Forwarder][2] instead of the Datadog Lambda Extension.

Reviewer: ```suggestion
If your Go Lambda functions are still using runtime `go1.x`, consider either [migrating][3] to `provided.al2` or using the [Datadog Forwarder][2] instead of the Datadog Lambda Extension.
```

Follow these steps to instrument the function:

1. Set the environment variable `DD_API_KEY` to your Datadog API key from [API Management][5].

> 1. Set environment variable `DD_TRACE_ENABLED` to `true`.

Reviewer: ```suggestion
1. Set the environment variable `DD_TRACE_ENABLED` to `true`.
```

### Tag

> Although it's optional, Datadog highly recommends tagging you serverless applications with the `env`, `service`, and `version` tags following the [unified service tagging documentation][6].

Reviewer: ```suggestion
Optionally, tag your serverless applications with `env`, `service`, and `version`. For more information, see the [Unified Service Tagging documentation][6].
```

}
```

> For more information on custom metric submission, see [here][8].

Reviewer: ```suggestion
For more information, see the [Custom Metrics documentation][8].
```
